### PR TITLE
enhancement(PeriodFilter): improve behaviour of end date

### PR DESCRIPTION
**Pull request by ersachin3112**
Resolve https://github.com/opencollective/opencollective/issues/4424

# Description
improve behaviour of end date

# Screenshots
Screenshot 2021-07-30 at 11 25 54 AM

### CODE REVIEW

File: components/budget/filters/PeriodFilter.js [hunk 1]
```diff
@@ -119,7 +123,7 @@ const TriggerContainer = styled(StyledButton)`
 const PeriodFilter = ({ onChange, value, inputId, ...props }) => {
   const [dateInterval, setDateInterval] = React.useState(() => getDefaultState(value));
   const setDate = (type, date) => {
-    setDateInterval(value => ({ ...value, [type]: !date ? null : normalizeDate(new Date(date)) }));
+    return setDateInterval(value => ({ ...value, [type]: !date ? null : normalizeDate(new Date(date), type) }));
```
Comment (Betree) [inline components/budget/filters/PeriodFilter.js:126]: Why this return? The return value doesn't seem to be used

**Review by Betree — CHANGES_REQUESTED** [review]
Sorry for the late review, this works great! However there is a regression: the inputs are not pre-filled with the date values like they used to. We probably need to truncate the time when passing `value`, or turn the inputs into datetime ones.

Peek 2021-08-17 18-38

**Comment by Betree** [comment]
@ersachin3112 any news on this one? Can I help with anything?

File: components/budget/filters/PeriodFilter.js (hunk 1 above)
Comment (ersachin3112) [reply components/budget/filters/PeriodFilter.js]: Yes, we can remove it.

**Comment by ersachin3112** [comment]
> Sorry for the late review, this works great! However there is a regression: the inputs are not pre-filled with the date values like they used to. We probably need to truncate the time when passing value, or turn the inputs into datetime ones.

@Betree: Sorry for delay, I think this regression is not related to this PR as the inputs are not pre-filled on `production` and `staging` as well. Can you verify?  See at https://opencollective.com/babel/transactions .

**Comment by Betree** [comment]
Closing in favor of https://github.com/opencollective/opencollective-frontend/pull/6887

**Pull request closed without merging**